### PR TITLE
Move composer dependencies for Javascript plugins to the root of the project

### PR DESCRIPTION
**Pull request by nuez**
Instead of having composer dependencies in the themes, move them to the root of the project. Alternative approach would be to turn themes into separate repositories, and add those a depencency in composer.json of root.

I think the latter would be overkill. 

This PR puts the dependencies in composer.json of the project root. Next problem: the dependencies are frontend javascript libraries, which are now located in /vendor/..., whereas the website resides in /web. /vendor/ therefor has to be web-accesible.

@NITEMAN Would that work? Any ideas?

### CODE REVIEW

**Comment by NITEMAN** [comment]
@nuez you can use extra[installer-paths] in composer.json to provide a custom destination (see current composer.json for a type based example)

**Comment by NITEMAN** [comment]
BTW, javascript dependencies use to be managed outside composer with things like bower/gulp/grunt/etc. deployment process can be tailored to use such tools

**Comment by nuez** [comment]
Thanks @NITEMAN I'll look into your suggestions

**Comment by NITEMAN** [comment]
@nuez don't forget to update ansible/deploy.yaml to remove unneded composer calls

**Comment by nuez** [comment]
OK, i've ended up including the Javascript files directly into the repo, using D8 libraries API. Libraries are declared in the dcamp_base_theme.

1. extra[installer-paths] only works for packaged libraries (not just github repos) as you can read on: https://getcomposer.org/doc/faqs/how-do-i-install-a-package-to-a-custom-path-for-my-framework.md 

> Note: You cannot use this to change the path of any package. This is only applicable to packages that require composer/installers and use a custom type that it handles.

2. Not all libraries used by the theme are packaged in bower, so that's not an option either, 

Composer calls in ansible/deploy.yml have been removed.

File: composer.json [hunk 1]
```diff
@@ -1,65 +1,74 @@
...
-            "name": "",
-            "role": ""
-        }
-    ],
-    "repositories": [
-        {
-            "type": "composer",
-            "url": "https://packagist.drupal-composer.org"
-        }
+  "name": "drupal-composer/drupal-project",
```
Comment (juampynr) [inline composer.json]: As far as I know , this is obsolete. We should follow what https://www.drupal.org/node/2404989 says. Thoughts @nuez?

Comment (NITEMAN) [inline composer.json]: AFAIK D.O packagist still have some rough edges. I'll try to move forward an discuss about packagist in another issue

**Comment by nuez** [comment]
I agree with @NITEMAN that we can discuss packagist's rough edges in another issue,
The link you suggest @juampynr can be part of the new issue. 
This PR is merely about solving the JS dependencies for the themes, and if the solution is satisfactory I think we should merge it.

**Pull request merged**